lib/ssl/src/ssl_handshake.erl

```diff
@@ -1151,6 +1202,30 @@ maybe_add_tls13_extensions({3,4},
 maybe_add_tls13_extensions(_, HelloExtensions, _, _, _) ->
     HelloExtensions.
 
+add_ocsp_stapling_extensions(
+    _Version, #{ocsp_stapling        := OcspStapling,
+                ocsp_responder_certs := OcspResponderCerts},
+    OcspNonce, HelloExtensions) ->
+    case OcspStapling of
+        false ->
+            HelloExtensions;
+        true ->
+            OcspResponderList = get_ocsp_responder_list(OcspResponderCerts, []),
+            OcspRequestExtns = public_key:ocsp_extensions(OcspNonce),
+            Req = #ocsp_status_request{responder_id_list = OcspResponderList,
+                                       request_extensions = OcspRequestExtns},
+            CertStatusReqExtn = #certificate_status_request{
+                status_type = ?CERTIFICATE_STATUS_TYPE_OCSP,
+                request = Req
+            },
+            HelloExtensions#{ocsp_stapling => CertStatusReqExtn}
+    end.
+
```

Comment:
I would call this function maybe_add_certificate_status_request to follow the pattern. `maybe_` expresses that there is a condition for adding the extension. As there is only one OCSP extension defined by RFC6066 it is easier to understand if we name that specific extension.

Also it could be factored into two separate clauses:
```erlang
maybe_add_certificate_status_request(_Version, #{ocsp_stapling := false}, _OcspNonce, HelloExtensions) ->
    HelloExtensions;
maybe_add_certificate_status_request(
    _Version, #{ocsp_stapling := true,
                ocsp_responder_certs := OcspResponderCerts}, _OcspNonce, HelloExtensions) ->
...
```

Reply:
done

Reply:
> I would call this function maybe_add_certificate_status_request to follow the pattern. `maybe_` expresses that there is a condition for adding the extension. As there is only one OCSP extension defined by RFC6066 it is easier to understand if we name that specific extension.
> 
> Also it could be factored into two separate clauses:
> 
> ```erlang
> maybe_add_certificate_status_request(_Version, #{ocsp_stapling := false}, _OcspNonce, HelloExtensions) ->
>     HelloExtensions;
> maybe_add_certificate_status_request(
>     _Version, #{ocsp_stapling := true,
>                 ocsp_responder_certs := OcspResponderCerts}, _OcspNonce, HelloExtensions) ->
> ...
> ```

done